connect/runtime/src/test/java/org/apache/kafka/connect/util/clusters/EmbeddedConnectStandalone.java

```diff
@@ -0,0 +1,140 @@
+
+    private final Map<String, String> workerProps;
+    private final String offsetsFile;
+
+    private WorkerHandle connectWorker;
+
+    private EmbeddedConnectStandalone(
+            int numBrokers,
+            Properties brokerProps,
+            boolean maskExitProcedures,
+            Map<String, String> clientProps,
+            Map<String, String> workerProps,
+            String offsetsFile
+    ) {
+        super(numBrokers, brokerProps, maskExitProcedures, clientProps);
+        this.workerProps = workerProps;
+        this.offsetsFile = offsetsFile;
+    }
+
+    @Override
+    public void startConnect() {
+        log.info("Starting standalone Connect worker");
+
+        workerProps.put(BOOTSTRAP_SERVERS_CONFIG, kafka().bootstrapServers());
+        // use a random available port
+        workerProps.put(LISTENERS_CONFIG, "HTTP://" + REST_HOST_NAME + ":0");
+
+        workerProps.putIfAbsent(OFFSET_STORAGE_FILE_FILENAME_CONFIG, offsetsFile);
+        workerProps.putIfAbsent(KEY_CONVERTER_CLASS_CONFIG, "org.apache.kafka.connect.storage.StringConverter");
+        workerProps.putIfAbsent(VALUE_CONVERTER_CLASS_CONFIG, "org.apache.kafka.connect.storage.StringConverter");
+        workerProps.putIfAbsent(PLUGIN_DISCOVERY_CONFIG, "hybrid_fail");
+
+        Connect connect = new ConnectStandalone().startConnect(workerProps);
+        connectWorker = new WorkerHandle("standalone", connect);
+    }
+
+    @Override
+    public String toString() {
+        return String.format("EmbeddedConnectStandalone(numBrokers= %d, workerProps= %s)",
+            numBrokers,
+            workerProps);
+    }
+
+    @Override
+    protected Set<WorkerHandle> workers() {
+        return connectWorker != null
+                ? Collections.singleton(connectWorker)
+                : Collections.emptySet();
+    }
+
+    public static class Builder extends EmbeddedConnectBuilder<EmbeddedConnectStandalone, Builder> {
+
+        private String offsetsFile = null;
+
+        public Builder offsetsFile(String offsetsFile) {
+            this.offsetsFile = offsetsFile;
+            return this;
+        }
+
+        @Override
+        protected EmbeddedConnectStandalone build(
+                int numBrokers,
+                Properties brokerProps,
+                boolean maskExitProcedures,
+                Map<String, String> clientProps,
+                Map<String, String> workerProps
+        ) {
+            if (offsetsFile == null)
+                offsetsFile = tempOffsetsFile();
+
+            return new EmbeddedConnectStandalone(
+                    numBrokers,
+                    brokerProps,
+                    maskExitProcedures,
+                    clientProps,
+                    workerProps,
+                    offsetsFile
+            );
+        }
+
+        private String tempOffsetsFile() {
+            try {
+                return TestUtils
+                        .tempFile("connect-standalone-offsets", "")
```

Comment:
nit: you can specify null to get the default `.tmp` which prevents the filename randomness from becoming the file extension.
```suggestion
                        .tempFile("connect-standalone-offsets", null)
```

Reply:
Was afraid to apply directly via the UI after rebasing onto trunk locally, but I did apply this in a follow-up commit that I've pushed now. Good catch, thanks!